File: VB/XPOInstant.SL.Web/XPOService.svc.vb
﻿Imports System
Imports System.Configuration
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB

Namespace XPOService
    Public Class WcfXpoSCService
        Inherits DataStoreService

        Private Shared ReadOnly Provider As IDataStore = XpoDefault.GetConnectionProvider(ConfigurationManager.ConnectionStrings("XpoConnection").ConnectionString, AutoCreateOption.DatabaseAndSchema)
        Shared Sub New()
        End Sub
        Public Sub New()
            MyBase.New(WcfXpoSCService.Provider)
        End Sub
    End Class
End Namespace

File: CS/XPOInstant.SL.Web/XPOService.svc.cs
﻿using System;
using System.Configuration;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;

namespace XPOService {
    public class WcfXpoSCService: DataStoreService {
        private static readonly IDataStore ConnectionProvider = XpoDefault.GetConnectionProvider(ConfigurationManager.ConnectionStrings["XpoConnection"].ConnectionString, AutoCreateOption.DatabaseAndSchema);
        static WcfXpoSCService() { }
        public WcfXpoSCService() : base(WcfXpoSCService.ConnectionProvider) { }
    }
}